### PR TITLE
Attempt to allow torch 1.10 again

### LINKED ISSUE #3014: [bug] Memory leak on GPU
### Issue Description
After successfully following the SS-VAE example in pyro I have moved over to training a variant of it on a GPU. A memory leak has arisen that has been boiled down to a minimum working example that replicates the error. I am at a loss as to how to proceed in rectifying this issue, any help or insight would be much appreciated!

### Environment
 - OS and python version. 
   - OS: Red Hat Enterprise Linux Workstation release 7.9 (Maipo) 
   - Python: 3.9.7
   - CUDA Version: 11.2
 - PyTorch version: 1.10.0
 - Pyro version: 1.7.0

### Code Snippet
``` python
from datetime import datetime

import pyro
import pyro.distributions as dist
import torch
import torch.nn as nn
from pyro.infer import SVI, TraceEnum_ELBO
from pyro.optim import Adam


class SSVAE(nn.Module):
    def __init__(self):
        super().__init__()
        self.accept_ratio = .5
        ones = torch.ones((), device='cuda')
        ratio = 1 / (self.accept_ratio + 1)
        self.alpha_prior = torch.stack((ones - ratio, ones - self.accept_ratio * ratio)).T
        self.cuda()

    def model(self):
        pyro.sample("y", dist.OneHotCategorical(self.alpha_prior))

    def guide(self):
        pass


def sizeof_fmt(num, suffix="B"):
    for unit in ["", "Ki", "Mi", "Gi", "Ti", "Pi", "Ei", "Zi"]:
        if abs(num) < 1024.0:
            return f"{num:3.1f}{unit}{suffix}"
        num /= 1024.0
    return f"{num:.1f}Yi{suffix}"


if __name__ == '__main__':
    torch.cuda.empty_cache()
    ssvae = SSVAE()

    optimizer = Adam({"lr": 1e-4})

    svi = SVI(ssvae.model, ssvae.guide, optimizer, loss=TraceEnum_ELBO())

    for i in range(10):
        print(f"{datetime.now().strftime('%H:%M:%S')} - Starting epoch #{i + 1}")
        print(
            f"Current GPU memory allocation:"
            f" {sizeof_fmt(torch.cuda.memory_allocated())}/{sizeof_fmt(torch.cuda.max_memory_allocated())}")

        # get the losses for an epoch
        for _ in range(100):
            svi.step()
```

### LINKED ISSUE #3056: Error messages during Pyro installation in Colab `torchvision 0.11.1+cu111 requires torch==1.10.0, but you have torch 1.11.0 which is incompatible.`
This might not be critical but right now when you install pyro-ppl-1.8.1 in google colab pip complains with error messages:

```
!pip install pyro-ppl

Collecting pyro-ppl
  Downloading pyro_ppl-1.8.1-py3-none-any.whl (718 kB)
     |████████████████████████████████| 718 kB 5.5 MB/s 
Collecting pyro-api>=0.1.1
  Downloading pyro_api-0.1.2-py3-none-any.whl (11 kB)
Requirement already satisfied: opt-einsum>=2.3.2 in /usr/local/lib/python3.7/dist-packages (from pyro-ppl) (3.3.0)
Collecting torch>=1.11.0
  Downloading torch-1.11.0-cp37-cp37m-manylinux1_x86_64.whl (750.6 MB)
     |████████████████████████████████| 750.6 MB 12 kB/s 
Requirement already satisfied: numpy>=1.7 in /usr/local/lib/python3.7/dist-packages (from pyro-ppl) (1.21.5)
Requirement already satisfied: tqdm>=4.36 in /usr/local/lib/python3.7/dist-packages (from pyro-ppl) (4.63.0)
Requirement already satisfied: typing-extensions in /usr/local/lib/python3.7/dist-packages (from torch>=1.11.0->pyro-ppl) (3.10.0.2)
Installing collected packages: torch, pyro-api, pyro-ppl
  Attempting uninstall: torch
    Found existing installation: torch 1.10.0+cu111
    Uninstalling torch-1.10.0+cu111:
      Successfully uninstalled torch-1.10.0+cu111
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
torchvision 0.11.1+cu111 requires torch==1.10.0, but you have torch 1.11.0 which is incompatible.
torchtext 0.11.0 requires torch==1.10.0, but you have torch 1.11.0 which is incompatible.
torchaudio 0.10.0+cu111 requires torch==1.10.0, but you have torch 1.11.0 which is incompatible.
Successfully installed pyro-api-0.1.2 pyro-ppl-1.8.1 torch-1.11.0
```

### PR DESCRIPTION
**Pull request by fritzo**
Resolves #3014, #3068

If Pyro still works with PyTorch 1.10, we can relax the version requirement and re-release Pyro 1.8.2 to save colab users a headache.

### CODE REVIEW

**Comment by martinjankowiak** [comment]
@fritzo this can be closed, right?

**Comment by fritzo** [comment]
@martinjankowiak I still aspire to get this working. It appears PyTorch 1.11 leaks memory #3014 #3068, and until a fix is released I'd like to support PyTorch 1.10.
